common/src/main/java/com/wynntils/overlays/minecraft/HeldItemOverlay.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.overlays.minecraft;
+
+import com.mojang.blaze3d.platform.Window;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.consumers.overlays.MinecraftOverlay;
+import com.wynntils.core.consumers.overlays.Overlay;
+import com.wynntils.core.consumers.overlays.OverlayPosition;
+import com.wynntils.core.consumers.overlays.OverlaySize;
+import com.wynntils.core.persisted.Persisted;
+import com.wynntils.core.persisted.config.Config;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.mc.event.RenderEvent;
+import com.wynntils.mc.event.TickEvent;
+import com.wynntils.utils.colors.CommonColors;
+import com.wynntils.utils.render.buffered.BufferedFontRenderer;
+import com.wynntils.utils.render.type.HorizontalAlignment;
+import com.wynntils.utils.render.type.TextShadow;
+import com.wynntils.utils.render.type.VerticalAlignment;
+import com.wynntils.utils.wynn.InventoryUtils;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class HeldItemOverlay extends Overlay implements MinecraftOverlay {
+    @Persisted
+    public final Config<TextShadow> textShadow = new Config<>(TextShadow.NORMAL);
+
+    @Persisted
+    public final Config<Float> fontScale = new Config<>(1.0f);
+
+    @Persisted
+    public final Config<Boolean> hideHeldItemName = new Config<>(false);
+
+    private static final int MESSAGE_TICKS = 40;
+
+    private int messageTimer;
+    private ItemStack heldItem = ItemStack.EMPTY;
+    private StyledText message;
+
+    public HeldItemOverlay() {
+        // It's not 100% exact but it's close enough to the vanilla position
+        super(
+                new OverlayPosition(
+                        -45,
+                        0,
+                        VerticalAlignment.BOTTOM,
+                        HorizontalAlignment.CENTER,
+                        OverlayPosition.AnchorSection.BOTTOM_MIDDLE),
+                new OverlaySize(200, 20),
+                HorizontalAlignment.CENTER,
+                VerticalAlignment.MIDDLE);
+    }
+
+    @SubscribeEvent
+    public void onRender(RenderEvent.Pre event) {
+        // Always cancel vanilla rendering
+        if (event.getType() == RenderEvent.ElementType.HELD_ITEM_NAME) {
+            event.setCanceled(true);
+        }
+    }
+
+    @SubscribeEvent
+    public void onTick(TickEvent event) {
+        ItemStack currentItem = InventoryUtils.getItemInHand();
+
+        if (!ItemStack.matches(currentItem, heldItem)) {
+            heldItem = currentItem;
+
+            if (currentItem.getItem() == Items.AIR) {
+                message = StyledText.fromString("");
+            } else {
+                // FIXME: Italics don't render
```

Comment:
So, how come :)? What are we talking about? Those custom named items, that are made italic by Minecraft?

Reply:
I suppose "don't render" isn't a good description of the issue. `heldItem.getHoverName()` just doesn't return the `§o`, it was most noticeable on soul points, though we won't have to worry about that for long, but it was also missing the colour for the soul point count. the 15 is meant to be yellow and italics but it's just white.

I also had some other item that was just italics and one colour but the italics still didn't work

Reply:
Top is overlay, bottom vanilla
![image](https://github.com/Wynntils/Artemis/assets/8337467/94967448-ba23-41f6-93fa-4a223fc6c983)
![image](https://github.com/Wynntils/Artemis/assets/8337467/bad64e49-9e1b-44bf-b192-05b43c4d3516)